torch/nested/_internal/ops.py

```diff
@@ -1471,6 +1471,58 @@ def slice_tensor(func, *args, **kwargs):
     return NestedTensor(func(inp._values, **new_kwargs), **extract_kwargs(inp))
 
 
+@register_jagged_func(
+    torch.ops.aten.index_put.default,
+    "input: jt_all, indices: any, values: t, accumulate: any?",
+)
+@register_jagged_func(
+    torch.ops.aten.index_put_.default,
+    "input: jt_all, indices: any, values: t, accumulate: any?",
+)
+def index_put_(func, *args, **kwargs):
+    _, new_kwargs = normalize_function(  # type: ignore[misc]
+        func, args=args, kwargs=kwargs, normalize_to_only_use_kwargs=True
+    )
+
+    inp: NestedTensor = new_kwargs.pop("input")
+
+    # For index_put_ to work, we add together the indices of the ragged dimension
+    # and the batch dimension right before it, changing the batch indices by
+    # the offsets of each ragged dimension
+
+    indices = new_kwargs.pop("indices")
```

Comment:
my understanding is that, for dense tensors, `indices` is expected to be a tuple (or list) of tensors with length in `[0, t.dim()]`. It can be an empty tuple, in which case no indices are updated. It can also have length < `t.dim()` for operating on bigger slices.

I think we need some verification / different handling on the length of the `indices` tuple since right now I can trigger `IndexError: list index out of range` by passing too small of an indices tuple, and that shouldn't happen